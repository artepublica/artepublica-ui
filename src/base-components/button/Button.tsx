import { GestureResponderEvent, Pressable, StyleSheet, Text } from 'react-native';

import { Theme, useTheme } from '@utils';

export type ButtonProps = {
  children?: string;
  onPress?: (event: GestureResponderEvent) => void;
  testOnly_pressed?: boolean;
};

function Button({ children, onPress, testOnly_pressed }: ButtonProps): JSX.Element {
  const { theme } = useTheme();

  const style = styles(theme);

  const button = (
    <Pressable
      onPress={onPress}
      style={({ pressed }) => [style.button, pressed && style.pressed]}
      testOnly_pressed={testOnly_pressed}
    >
      <Text>{children?.toLowerCase()}</Text>
    </Pressable>
  );

  return button;
}

function styles(theme: Theme) {
  return StyleSheet.create({
    button: {
      fontFamily: theme.button.font.fontFamily,
      color: theme.button.textColor,
      borderWidth: 1,
      borderColor: theme.button.borderColor,
      borderRadius: 8,
      backgroundColor: theme.button.backgroundColor,
      padding: 16,
    },
    pressed: {
      color: theme.button.pressed.textColor,
      borderColor: theme.button.pressed.borderColor,
      backgroundColor: theme.button.pressed.backgroundColor,
    },
  });
}

export default Button;
