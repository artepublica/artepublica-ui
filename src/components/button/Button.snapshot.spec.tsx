import { create } from 'react-test-renderer';

import Button from './Button';

describe('<Button />', () => {
    it('default snapshot', () => {
        const title = 'test';
        const json = create(<Button>{title}</Button>).toJSON();

        expect(json).toMatchSnapshot();
    });

    it('pressed snapshot', () => {
        const title = 'test';
        const json = create(<Button testOnly_pressed>{title}</Button>).toJSON();

        expect(json).toMatchSnapshot();
    });
});
