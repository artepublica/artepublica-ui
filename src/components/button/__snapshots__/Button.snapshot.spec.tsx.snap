// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> default snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderColor": "white",
        "borderRadius": 8,
        "borderWidth": 1,
        "color": "white",
        "fontFamily": "fontFamily",
        "padding": 16,
      },
      false,
    ]
  }
>
  <Text>
    test
  </Text>
</View>
`;

exports[`<Button /> pressed snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderColor": "white",
        "borderRadius": 8,
        "borderWidth": 1,
        "color": "white",
        "fontFamily": "fontFamily",
        "padding": 16,
      },
      Object {
        "backgroundColor": "black",
        "borderColor": "black",
        "color": "black",
      },
    ]
  }
>
  <Text>
    test
  </Text>
</View>
`;
